fsharp-backend/src/LibExecutionStdLib/LibTuple.fs

```diff
@@ -0,0 +1,348 @@
+module LibExecutionStdLib.LibTuple
+
+open Prelude
+open LibExecution.RuntimeTypes
+open LibExecution.VendoredTablecloth
+
+module Interpreter = LibExecution.Interpreter
+
+module Errors = LibExecution.Errors
+
+let fn = FQFnName.stdlibFnName
+
+let err (str : string) = Ply(Dval.errStr str)
+
+let incorrectArgs = Errors.incorrectArgs
+
+// TODO consider breaking this into `tuple2fns` and `tuple3fns` in-code,
+// and then `let fns = tuple2fns @ tuple3fns. (may help with legibility?)
+let fns : List<BuiltInFn> =
+  [ // Tuple2
+    { name = fn "Tuple2" "pair" 0
+      parameters =
+        [ Param.make "first" (TVariable "a") ""
+          Param.make "second" (TVariable "b") "" ]
+      returnType = TTuple(TVariable "a", TVariable "b", [])
+      description = "Returns a new 2-tuple with the given values."
+      fn =
+        (function
+        | state, [ first; second ] -> Ply(DTuple(first, second, []))
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "Tuple2" "first" 0
+      parameters =
+        [ Param.make "tuple" (TTuple(TVariable "a", TVariable "b", [])) "" ]
+      returnType = TVariable "a"
+      description = "Returns the first value of a 2-tuple."
+      fn =
+        (function
+        | state, [ DTuple (first, _second, []) ] -> Ply(first)
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "Tuple2" "second" 0
+      parameters =
+        [ Param.make "tuple" (TTuple(TVariable "a", TVariable "b", [])) "" ]
+      returnType = TVariable "b"
+      description = "Returns the second value of a 2-tuple."
+      fn =
+        (function
+        | state, [ DTuple (_first, second, []) ] -> Ply(second)
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "Tuple2" "swap" 0
+      parameters =
+        [ Param.make "tuple" (TTuple(TVariable "a", TVariable "b", [])) "" ]
+      returnType = TTuple(TVariable "b", TVariable "a", [])
+      description = "Returns a 2-tuple with the elements swapped."
+      fn =
+        (function
+        | state, [ DTuple (first, second, []) ] -> Ply(DTuple(second, first, []))
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "Tuple2" "mapFirst" 0
+      parameters =
+        [ Param.makeWithArgs
+            "fn"
+            (TFn([ TVariable "a" ], TVariable "c"))
```

Comment:
Would we do better with descriptive names instead of "a" and "c"? Nothing comes to mind personally.

Reply:
I thought about using `first` etc. instead of `a`, but felt it to be a bit much. We can always adjust later.